xclim/sdba/adjustment.py

```diff
@@ -1204,6 +1245,253 @@ def _adjust(
+                    f"`crd_dim` attribute of `rot_matrices` ({rot_matrices.attrs['crd_dim']}) does not correspond to `pts_dim` ({pts_dim})."
+                )
+        else:
+            rot_dim = xr.core.utils.get_temp_dimname(
+                set(ref.dims).union(hist.dims), pts_dim + "_prime"
+            )
+            rot_matrices = rand_rot_matrix(
+                ref[pts_dim], num=n_iter, new_dim=rot_dim
+            ).rename(matrices="iterations")
+        pts_dims = [rot_matrices.attrs[d] for d in ["crd_dim", "new_dim"]]
+
+        # time indices corresponding to group and windowed group
+        # used to divide datasets as map_blocks or groupby would do
+        g_idxs, gw_idxs = grouped_time_indexes(ref.time, base_kws["group"])
+
+        # training, obtain adjustment factors of the npdf transform
+        ds = xr.Dataset(dict(ref=ref, hist=hist))
+        params = {
+            "quantiles": base_kws["nquantiles"],
+            "interp": adj_kws["interp"],
+            "extrapolation": adj_kws["extrapolation"],
+            "pts_dims": pts_dims,
+            "n_escore": n_escore,
+        }
+        out = mbcn_train(
+            ds, rot_matrices=rot_matrices, g_idxs=g_idxs, gw_idxs=gw_idxs, **params
+        )
+        params["group"] = base_kws["group"]
+
+        # postprocess
+        out["rot_matrices"] = rot_matrices
+        out["g_idxs"] = g_idxs
+        out["gw_idxs"] = gw_idxs
+
+        out.af_q.attrs.update(
+            standard_name="Adjustment factors",
+            long_name="Quantile mapping adjustment factors",
+        )
+        return out, params
+
+    def _adjust(
+        self,
+        sim: xr.DataArray,
+        ref: xr.DataArray,
+        hist: xr.DataArray,
+        *,
+        base: TrainAdjust = QuantileDeltaMapping,
+        base_kws_vars: dict[str, Any] | None = None,
+        adj_kws: dict[str, Any] | None = None,
+        period_dim=None,
+    ):
+        # set default values for non-specified parameters
+        base_kws_vars = base_kws_vars or {}
+        for v in sim[self.pts_dims[0]].values:
+            if v not in base_kws_vars.keys():
+                base_kws_vars[v] = {}
+
+            base_kws_vars[v].setdefault("group", self.group)
+            if isinstance(base_kws_vars[v]["group"], str):
+                base_kws_vars[v]["group"] = Grouper(base_kws_vars[v]["group"], 1)
+            if base_kws_vars[v]["group"] != self.group:
+                raise ValueError(
+                    "Expected usage of the same group in adjust and training steps."
+                )
+            base_kws_vars[v].pop("group")
+
+            base_kws_vars[v].setdefault("nquantiles", self.ds.af_q.quantiles.values)
+            if np.isscalar(base_kws_vars[v]["nquantiles"]):
+                base_kws_vars[v]["nquantiles"] = equally_spaced_nodes(
+                    base_kws_vars[v]["nquantiles"]
+                )
+            # should maybe require that quantiles here match those from training?
```

Comment:
`group` must be the same as in training, but otherwise different training/adjusting arguments can be given for the QDM and npdf.  To simplify things, I could require that we must use the same arguments. I would be a bit confusing because all `adj_kws`  would then be passed to the train function of `MBCn`